spec/inputs/irc_spec.rb

```diff
@@ -1,3 +1,72 @@
 # encoding: utf-8
-require "logstash/devutils/rspec/spec_helper"
-require 'logstash/inputs/irc'
+require_relative "../spec_helper"
+require "logstash/plugin"
+require "logstash/event"
+require "cinch"
+
+describe LogStash::Inputs::Irc do
+
+  let(:properties) { {:name => "foo" } }
+  let(:event)      { LogStash::Event.new(properties) }
+
+  let(:host)       { "irc.freenode.org" }
+  let(:channels)   { ["foo", "bar"] }
+
+  it "should register without errors" do
+    plugin = LogStash::Plugin.lookup("input", "irc").new({ "host" => host, "channels" => channels })
+    expect { plugin.register }.to_not raise_error
+  end
+
+  describe "receive" do
```

Comment:
Am I correct in thinking that the input helper calls input.run(q) which then calls bot.run which tries to connect to the `irc.freenode.org`?  If I am, then this is why I asked whether you were mocking.

I expected, that for a unit test like this, the bot dependency did not need to make a connection to the internet because you are simulating responses.